### PR TITLE
Add Serialisation Support

### LINKED ISSUE #25: Serialising KmerCountTables
Serialise `KmerCountTable` objects to disk and load saved objects.

Check out [Serde](https://serde.rs/#data-formats) and [Bincode](https://github.com/bincode-org/bincode).

Serde supports read/write of Python Pickle format. Could make things easier when loading back into Python.

### PR DESCRIPTION
**Pull request by Adamtaranto**
Add serialisation support for `KmerCountTable` objects using Serde.

New methods: 
- `.serialize_json()` this will serialise a count table to JSON format, returns raw JSON string. Mostly useful for testing.
- `.save("counts.json.gz")` this will convert a count table to JSON format, compress it with gzip, and write to a target file.
- `.load()` is a static method that will load a count table saved in `json.gz` format and return a new `KmerCountTable` with the loaded properties. Raises a warning if Oxli version of the saved object is different to the current version. 

Closes #25

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -14,3 +14,10 @@ sourmash = "0.15.1"
 anyhow = "1.0.89"
 log = "0.4.22"
 env_logger = "0.11.5"
+
+# For JSON serialization/deserialization
+serde = { version = "1.0", features = ["derive"] }
```
Comment (ctb) [inline Cargo.toml:19]: why version 1? latest is 1.0.210. (presumably dependabot will upgrade, just curious.)

File: Cargo.toml [hunk 2]
```diff
@@ -14,3 +14,10 @@ sourmash = "0.15.1"
...
+serde_json = "1.0"
+
+# For Gzip compression/decompression
+flate2 = "1.0"
```
Comment (ctb) [inline Cargo.toml:23]: suggest niffler, which allows sniffing/auto-determination of file formats. but we can backport that in if we need.

(I'm using it over in #10)

**Review by ctb — APPROVED** [review]

File: Cargo.toml (hunk 1 above)
Comment (Adamtaranto) [reply Cargo.toml:19]: No good reason, just inherited from example code.

**Comment by Adamtaranto** [comment]
Hey @ctb, can you look over the niffler changes. It passes all my tests, but the code is a bit rough, especially with the error handling. 

Any additional test case suggestion welcome.

**Review by ctb — APPROVED** [review]
looks good to me! great work! I updated a few things & added tests here: https://github.com/oxli-bio/oxli/pull/49.

**Pull request merged**